Dependency Tree issue

### Issue
Encountered an error while trying to install package on project. The error prompted was the following one: 

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: project@undefined
npm ERR! Found: @contentful/rich-text-types@16.3.0
npm ERR! node_modules/@contentful/rich-text-types
npm ERR!   @contentful/rich-text-types@"^16.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @contentful/rich-text-types@"^15.14.1" from contentful-rich-text-vue-renderer@3.1.0
npm ERR! node_modules/contentful-rich-text-vue-renderer
npm ERR!   contentful-rich-text-vue-renderer@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```
Downgrading `@contentful/rich-text-types` to `@^15` allowed me to install the package.
After having a look to the changelog of the different rich-text related packages, it seems that the upgrade between 15 and 16 is only a version bump. Would you consider updating the dependencies on this package ? 

### Project Package.json

```json
  "devDependencies": {
    "@nuxt/devtools": "^0.8.4",
    "@nuxt/image": "^1.0.0-rc.1",
    "@nuxt/kit": "npm:@nuxt/kit-edge@latest",
    "@nuxtjs/eslint-config-typescript": "^12.0.0",
    ...
  },
  "dependencies": {
    "@contentful/rich-text-html-renderer": "^16.0.2",
    "@contentful/rich-text-types": "^16.0.2",
    "@pinia/nuxt": "^0.4.3",
    "@studio-freight/lenis": "^1.0.22",
    "contentful": "^10.5.1",
    "gsap": "^3.10.2",
    "h3-zod": "^0.5.1",
    "rupture": "^0.7.1",
    "zod": "^3.22.2"
  }
```

### Config
Darwin 22.6.0
node v18.18.0
npm  v9.8.1

### PS
Thanks for the package ✌️

--- a/README.md
+++ b/README.md
@@ -223,6 +223,9 @@ The `nodeRenderers` prop should be one of the following `BLOCKS` and `INLINES` p
   - `EMBEDDED_ENTRY`
   - `EMBEDDED_ASSET`
   - `TABLE`
+  - `TABLE_ROW`
+  - `TABLE_CELL`
+  - `TABLE_HEADER_CELL`
 
 - `INLINES`
   - `EMBEDDED_ENTRY` (this is different from the `BLOCKS.EMBEDDED_ENTRY`)
@@ -238,3 +241,4 @@ The `markRenderers` prop should be one of the following `MARKS` properties as de
 - `CODE`
 - `SUBSCRIPT`
 - `SUPERSCRIPT`
+- `STRIKETHROUGH`

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     "vue"
   ],
   "peerDependencies": {
-    "@contentful/rich-text-types": "^15.14.1"
+    "@contentful/rich-text-types": ">=16.5.0"
   },
   "peerDependenciesMeta": {
     "@contentful/rich-text-types": {
@@ -42,7 +42,7 @@
   "devDependencies": {
     "@babel/core": "^7.12.10",
     "@babel/preset-env": "^7.12.11",
-    "@contentful/rich-text-types": "^15.14.1",
+    "@contentful/rich-text-types": "^17.0.1",
     "@rollup/plugin-alias": "^3.1.9",
     "@rollup/plugin-babel": "^5.3.1",
     "@rollup/plugin-commonjs": "^22.0.0",

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     "vue"
   ],
   "peerDependencies": {
-    "@contentful/rich-text-types": ">=16.5.0"
+    "@contentful/rich-text-types": "^16.5.0 || ^17.0.0"
   },
   "peerDependenciesMeta": {
     "@contentful/rich-text-types": {

--- a/src/index.js
+++ b/src/index.js
@@ -24,6 +24,7 @@ const defaultMarkRenderers = {
   [MARKS.CODE]: (children, key) => h("code", { key }, children),
   [MARKS.SUPERSCRIPT]: (children, key) => h("sup", { key }, children),
   [MARKS.SUBSCRIPT]: (children, key) => h("sub", { key }, children),
+  [MARKS.STRIKETHROUGH]: (children, key) => h("s", { key }, children),
 };
 
 const defaultNodeRenderers = {

--- a/src/index.test.js
+++ b/src/index.test.js
@@ -49,6 +49,11 @@ describe("RichText", () => {
               value: "Subpar",
               marks: [{ type: MARKS.SUBSCRIPT }],
             },
+            {
+              nodeType: "text",
+              value: "Irrelevant",
+              marks: [{ type: MARKS.STRIKETHROUGH }],
+            },
           ],
         },
       ]);
@@ -56,7 +61,7 @@ describe("RichText", () => {
 
       it("renders them all in a single paragraph", () => {
         expect(rendered.html()).toBe(
-          '<p><strong>Hello</strong><em> world!</em><code>console.log("yo");</code><u>Greetings!</u><sup>Superb!</sup><sub>Subpar</sub></p>'
+          '<p><strong>Hello</strong><em> world!</em><code>console.log("yo");</code><u>Greetings!</u><sup>Superb!</sup><sub>Subpar</sub><s>Irrelevant</s></p>'
         );
       });
     });
@@ -83,11 +88,23 @@ describe("RichText", () => {
             { type: MARKS.SUBSCRIPT },
           ],
         },
+        {
+          nodeType: "text",
+          value: "!",
+          marks: [
+            { type: MARKS.UNDERLINE },
+            { type: MARKS.ITALIC },
+            { type: MARKS.BOLD },
+            { type: MARKS.STRIKETHROUGH },
+          ],
+        },
       ]);
       const rendered = mount(RichText, { props: { document } });
 
       it("renders all overlapping marks in order", () => {
-        expect(rendered.html()).toBe("<strong><em><u><sup>Hello</sup></u></em></strong>\n<strong><em><u><sub>World</sub></u></em></strong>");
+        expect(rendered.html()).toBe(
+          "<strong><em><u><sup>Hello</sup></u></em></strong>\n<strong><em><u><sub>World</sub></u></em></strong>\n<u><em><strong><s>!</s></strong></em></u>"
+        );
       });
     });
   });

